crates/remora/src/proxy/core.rs

```diff
@@ -238,28 +276,51 @@ mod tests {
             proxy_id, executor, mode, store, rx_proxy, tx_results, metrics,
         );
 
+        let transactions = if signed {
+            new_signed_txs().await
+        } else {
+            new_unsigned_txs().await
+        };
+
         // Send transactions to the proxy.
-        let transactions = generate_transactions(&config).await;
+        // let transactions = generate_transactions(&config).await;
         for tx in transactions {
-            let transaction = SuiTransaction::new_for_tests(tx);
-            tx_proxy.send(transaction).await.unwrap();
+            // let transaction = SuiTransaction::new_for_tests(tx);
+            tx_proxy.send(tx).await.unwrap();
         }
 
         // Spawn the proxy.
         proxy.spawn();
 
         // Receive the results.
         let results = rx_results.recv().await.unwrap();
-        assert!(results.success());
+        
+        if signed {
+            assert!(results.authentication_success);
+            assert!(results.success());
+        } else {
+            assert!(!results.authentication_success);
+            assert!(!results.success())
+        }
+    }
+
+    #[tokio::test]
+    async fn test_single_threaded_proxy_signed() {
+        pre_execute(ProxyMode::SingleThreaded, true).await;
+    }
+
+    #[tokio::test]
+    async fn test_multi_threaded_proxy_signed() {
+        pre_execute(ProxyMode::MultiThreaded, true).await;
```

Comment:
same question

Reply:
This test checks that the proxy correctly executes transactions if we run the proxy in **multi**-threaded mode and we send it transactions that are correctly signed.